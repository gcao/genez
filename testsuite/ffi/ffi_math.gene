# Test math FFI functions

# Declare sin from libm
(c-extern sin ([x "double"]) "double" "libm")

# Test sin with known values
(var pi_over_2 1.5707963267948966)
(var result (sin pi_over_2))

# sin(Ï€/2) should be 1.0
(print "sin(pi/2) = " result)

# Test sin(0) = 0
(var zero_result (sin 0))
(print "sin(0) = " zero_result)