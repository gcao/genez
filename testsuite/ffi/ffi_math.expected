sin(pi/2) =  1
sin(0) =  0